src/main/java/org/folio/circulation/domain/notice/TemplateRepository.java

```diff
@@ -0,0 +1,26 @@
+package org.folio.circulation.domain.notice;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.http.client.Response;
+
+public class TemplateRepository {
+
+  private TemplateRepository(
+    CollectionResourceClient templateNoticeClient) {
+    this.templateNoticesClient = templateNoticeClient;
+  }
+
+  private final CollectionResourceClient templateNoticesClient;
+
+  public static TemplateRepository using(Clients clients) {
+    return new TemplateRepository(clients.templateNoticeClient());
+  }
+
+  public CompletableFuture<Result<Response>> findById(String id) {
+    return templateNoticesClient.get(id);
+  }
```

Comment:
What benefit does this offer over using the template notice client directly?

Maybe have a method that returns whether the template exists or not instead.

Reply:
what if we remove repository and call templateNoticeClient directly from the handlers? failIfTemplateNotFound() I can move to validator

Reply:
I'm ok with removing the repository and making the client a dependency of a validator. 

I'm not sure this needs to be a validator, unless all invalid notices get deleted?

Reply:
done